### PR TITLE
Update runner version

### PR DESCRIPTION
**Pull request by sramekmichal**

### CODE REVIEW

File: src/aws.js [hunk 1]
```diff
@@ -11,18 +11,18 @@ function buildUserDataScript(githubRegistrationToken, label) {
...
       './run.sh',
     ];
   } else {
     return [
       '#!/bin/bash',
       'mkdir actions-runner && cd actions-runner',
       'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
-      'curl -O -L https://github.com/actions/runner/releases/download/v2.299.1/actions-runner-linux-${RUNNER_ARCH}-2.299.1.tar.gz',
-      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.299.1.tar.gz',
+      'curl -O -L https://github.com/actions/runner/releases/download/2.304.0/actions-runner-linux-${RUNNER_ARCH}-2.299.1.tar.gz',
```
Comment (markszabo) [inline src/aws.js:22]: Seems like you missed this one: 

```suggestion
      'curl -O -L https://github.com/actions/runner/releases/download/2.304.0/actions-runner-linux-${RUNNER_ARCH}-2.304.0.tar.gz',
```

**Comment by markszabo** [comment]
+1 for updating the runner version to the [latest](https://github.com/actions/runner/releases) version

File: src/aws.js [hunk 2]
```diff
@@ -11,18 +11,18 @@ function buildUserDataScript(githubRegistrationToken, label) {
       '#!/bin/bash',
       `cd "${config.input.runnerHomeDir}"`,
       'export RUNNER_ALLOW_RUNASROOT=1',
-      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
+      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label} --ephemeral`,
```
Comment (machulav) [inline src/aws.js:14]: Could you please elaborate on why `--ephemeral` is added? 
Have you been able to test this approach and received any improvements?

File: action.yml [hunk 3]
```diff
@@ -78,4 +78,4 @@ outputs:
       The id is used to terminate the EC2 instance when the runner is not needed anymore.
 runs:
   using: node16
-  main: ./dist/index.js
+  main: ./dist/index.js
```
Comment (machulav) [inline action.yml:81]: Please revert this change as it's not related.

**Review by machulav — CHANGES_REQUESTED** [review]
Hi @sramekmichal,

Thank you for your contribution!
May I ask you to look at my comments when you have time?

File: src/aws.js (hunk 2 above)
Comment (sramekmichal) [reply src/aws.js:14]: Hi @machulav,

it is related to this: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/autoscaling-with-self-hosted-runners#using-ephemeral-runners-for-autoscaling

File: src/aws.js [hunk 4]
```diff
@@ -11,18 +11,18 @@ function buildUserDataScript(githubRegistrationToken, label) {
...
     ];
   } else {
     return [
       '#!/bin/bash',
       'mkdir actions-runner && cd actions-runner',
       'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
-      'curl -O -L https://github.com/actions/runner/releases/download/v2.299.1/actions-runner-linux-${RUNNER_ARCH}-2.299.1.tar.gz',
-      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.299.1.tar.gz',
+      'curl -O -L https://github.com/actions/runner/releases/download/2.304.0/actions-runner-linux-${RUNNER_ARCH}-2.304.0.tar.gz',
+      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.304.0.tar.gz',
```
Comment (markszabo) [inline src/aws.js:23]: The current latest is `2.306.0`: https://github.com/actions/runner/releases/latest

```suggestion
      'curl -O -L https://github.com/actions/runner/releases/download/2.306.0/actions-runner-linux-${RUNNER_ARCH}-2.306.0.tar.gz',
      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.306.0.tar.gz',
```

**Comment by machulav** [comment]
fixed in #182
